File: FINHUB-TN-main/src/main/java/tn/finhub/model/ExpenseItem.java
package tn.finhub.model;

import java.math.BigDecimal;

public class ExpenseItem {
    private int id;
    private int expenseId;
    private String itemName;
    private BigDecimal price;

    public ExpenseItem() {
    }

    public ExpenseItem(int id, int expenseId, String itemName, BigDecimal price) {
        this.id = id;
        this.expenseId = expenseId;
        this.itemName = itemName;
        this.price = price;
    }

    // Getters and Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getExpenseId() {
        return expenseId;
    }

    public void setExpenseId(int expenseId) {
        this.expenseId = expenseId;
    }

    public String getItemName() {
        return itemName;
    }

    public void setItemName(String itemName) {
        this.itemName = itemName;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }
}

File: OneDrive/Desktop/FINHUB-WEB-main/src/Entity/ExpenseItem.php
<?php

namespace App\Entity;

use App\Repository\ExpenseItemRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ExpenseItemRepository::class)]
#[ORM\Table(name: 'expense_items')]
class ExpenseItem
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Expense::class)]
    #[ORM\JoinColumn(nullable: false, name: "expense_id", referencedColumnName: "id", onDelete: "CASCADE")]
    private ?Expense $expense = null;

    #[ORM\Column(length: 255)]
    private ?string $item_name = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 3)]
    private ?string $price = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getExpense(): ?Expense
    {
        return $this->expense;
    }

    public function setExpense(?Expense $expense): static
    {
        $this->expense = $expense;

        return $this;
    }

    public function getItemName(): ?string
    {
        return $this->item_name;
    }

    public function setItemName(string $item_name): static
    {
        $this->item_name = $item_name;

        return $this;
    }

    public function getPrice(): ?string
    {
        return $this->price;
    }

    public function setPrice(string $price): static
    {
        $this->price = $price;

        return $this;
    }
}
